File: ProjetScrapping/vol.py
# from selenium import webdriver
# from selenium.webdriver.firefox.options import Options
# from selenium.webdriver.common.by import By

# chrome_options = Options()
# chrome_options.add_argument('--headless')
# chrome_options.add_argument('--no-sandbox')
# chrome_options.add_argument('--disable-dev-shm-usage')

# driver = webdriver.Firefox(options=chrome_options)
# driver.get('https://booking.kayak.com/flights/DSS-PAR/2023-08-10?sort=price_a')

# elements = driver.find_elements(By.CLASS_NAME, 'flights')

# elementss = []

# for element in elements:
#     valeur = element.text
#     elementss.append(valeur)

# for element in elementss:
#     print(element)

# driver.quit()


# moreButton
# from selenium import webdriver
# from selenium.webdriver.firefox.options import Options
# from selenium.webdriver.common.by import By
# from selenium.webdriver.support.ui import WebDriverWait
# from selenium.webdriver.support import expected_conditions as EC

# chrome_options = Options()
# chrome_options.add_argument('--headless')
# chrome_options.add_argument('--no-sandbox')
# chrome_options.add_argument('--disable-dev-shm-usage')

# driver = webdriver.Firefox(options=chrome_options)
# driver.get('https://booking.kayak.com/flights/DSS-PAR/2023-08-10?sort=price_a')

# while True:
#     # selecs = driver.find_elements(By.CSS_SELECTOR, '.col-field.carrier')
#     # for selec in selecs:
#     #     elements = selec.find_element(By.CLASS_NAME, 'bottom')
#     #     print(elements.text)

#     selecs = driver.find_elements(By.CLASS_NAME, 'resultInner')
#     for selec in selecs:
#         # compagnie = selec.find_element(By.CSS_SELECTOR, '.col-field.carrier').find_element(By.CLASS_NAME, 'bottom')
#         # heuredep = selec.find_element(By.CSS_SELECTOR, '.col-field.time.depart').find_element(By.CLASS_NAME, 'top')
#         # aeroportdep = selec.find_element(By.CSS_SELECTOR, '.col-field.time.depart').find_element(By.CLASS_NAME, 'bottom')
#         # heurearr = selec.find_element(By.CSS_SELECTOR, '.col-field.time.return').find_element(By.CLASS_NAME, 'top')
#         # aeroportarr = selec.find_element(By.CSS_SELECTOR, '.col-field.time.return').find_element(By.CLASS_NAME, 'bottom')
#         prix = selec.find_element(By.CLASS_NAME, 'above-button').find_element(By.CLASS_NAME, 'price-text')
#         # element2 = selec.find_element(By.CLASS_NAME, 'bottom')
#         # print(element1.text)
#         print(prix.text)

#     try:
#         more_button = WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.CLASS_NAME, 'moreButton')))
#         more_button.click()
#         WebDriverWait(driver, 10).until(EC.staleness_of(selecs[-1]))
#     except:
#         break

# driver.quit()

import csv
from selenium import webdriver
from selenium.webdriver.firefox.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

chrome_options = Options()
chrome_options.add_argument('--headless')
chrome_options.add_argument('--no-sandbox')
chrome_options.add_argument('--disable-dev-shm-usage')

driver = webdriver.Firefox(options=chrome_options)
driver.get('https://booking.kayak.com/flights/DSS-PAR/2023-08-10?sort=price_a')

donnees = []

def scraping(x):
    compagnie = x.find_element(By.CSS_SELECTOR, '.col-field.carrier').find_element(By.CLASS_NAME, 'bottom').text
    heuredep = x.find_element(By.CSS_SELECTOR, '.col-field.time.depart').find_element(By.CLASS_NAME, 'top').text
    aeroportdep = x.find_element(By.CSS_SELECTOR, '.col-field.time.depart').find_element(By.CLASS_NAME, 'bottom').text
    heurearr = x.find_element(By.CSS_SELECTOR, '.col-field.time.return').find_element(By.CLASS_NAME, 'top').text
    aeroportarr = x.find_element(By.CSS_SELECTOR, '.col-field.time.return').find_element(By.CLASS_NAME, 'bottom').text
    prix = x.find_element(By.CLASS_NAME, 'above-button').find_element(By.CLASS_NAME, 'price-text').text
    
    donnees.append([compagnie,heuredep,aeroportdep,heurearr,aeroportarr,prix])


hasNextPage = True
pageNumber = 1

while hasNextPage & pageNumber <= 10 : 
    print('Page', pageNumber)

    driver.implicitly_wait(5)
    selecs = driver.find_elements(By.CLASS_NAME, 'resultInner')

    for selec in selecs:
        scraping(selec)
    
    nextPageButton = driver.find_element(By.CLASS_NAME, 'moreButton')
    hasNextPage = nextPageButton.is_enabled()

    if hasNextPage:
        driver.execute_script("arguments[0].click();", nextPageButton)
        # Attendre que la nouvelle page se charge
        driver.implicitly_wait(5)

        pageNumber += 1 

driver.quit()

File: ProjetScrapping/vol.js
const { Builder, By, Key, until } = require('selenium-webdriver');
const fs = require('fs');

async function initializeDriver() {
    let driver = await new Builder().forBrowser('firefox').build();
    return driver;
}

async function runScraping() {
    let driver = await initializeDriver();

    try {
        await driver.get('https://booking.kayak.com/flights/DSS-PAR/2023-08-10?sort=price_a');

        let hasNextPage = true;
        let pageNumber = 1;
        let tab = [];

        while (hasNextPage && pageNumber <= 300) {
            console.log('Page', pageNumber);

            // Attendre que les annonces se chargent
            await driver.wait(until.elementLocated(By.className('resultInner')), 10000);

            // Récupérer tous les éléments d'annonce
            let selecs = await driver.findElements(By.className('resultInner'));

            // Parcourir les annonces et extraire les informations souhaitées
            for (let selec of selecs) {
                let compagnie = await selec.findElement(By.className('col-field carrier')).findElement(By.className('bottom')).getText();
                let heuredep = await selec.findElement(By.className('col-field time depart')).findElement(By.className('top')).getText();
                let aeroportdep = await selec.findElement(By.className('col-field time depart')).findElement(By.className('bottom')).getText();
                let heurearr = await selec.findElement(By.className('col-field time return')).findElement(By.className('top')).getText();
                let aeroportarr = await selec.findElement(By.className('col-field time return')).findElement(By.className('bottom')).getText();
                let prix = await selec.findElement(By.className('above-button')).findElement(By.className('price-text')).getText();

                // console.log(compagnie, ' ', heuredep, ' ', aeroportdep, ' ', heurearr, ' ', aeroportarr, ' ', prix);

                tab.push({ compagnie, heuredep, aeroportdep, heurearr, aeroportarr, prix });
            }

            // Vérifier s'il y a une page suivante
            let nextPageButton = await driver.findElement(By.className('moreButton'));
            hasNextPage = await nextPageButton.isEnabled();

            // Si une page suivante existe, cliquer dessus
            if (hasNextPage) {
                await driver.executeScript("arguments[0].click();", nextPageButton);

                // Attendre que la nouvelle page se charge
                await driver.wait(until.elementLocated(By.className('resultInner')), 5000);
            }

            pageNumber++;
        }

        let jsonData = JSON.stringify(tab);

        fs.writeFile('data.json', jsonData, 'utf8', (err) => {
            if (err) {
                console.error('Une erreur s\'est produite lors de l\'écriture du fichier JSON :', err);
                return;
            }
            console.log('Les données ont été enregistrées dans le fichier JSON avec succès.');
        });
    } 
    finally {
        await driver.quit();
    }
}

runScraping();
